### ISSUE TITLE
Fix types for moonbeam v4 (testnet version 5)

### ISSUE THREAD

**Issue by joelamouche** [polkadot-js/api#2956]
Currently, when clicking a block on the explorer for the mooonbeam alphanet, this error is thrown: `Unable to retrieve the specified block details. createType(SignedBlock):: Struct: failed on block: {"header":"Header","extrinsics":"Vec<Extrinsic>"}:: Struct: failed on extrinsics: Vec<Extrinsic>:: createType(ExtrinsicV4):: createType(Call):: Call: failed decoding parachainUpgrade.setValidationData:: Struct: failed on args: :: Cannot construct unknown type ValidationData`

According to @JoshOrndorff the missing types are here: https://github.com/paritytech/polkadot/blob/master/primitives/src/v1.rs#L272
and the types should be injected this way https://github.com/polkadot-js/api/pull/2895/files because they would be useful for all parachains.

I made these:
```
export interface ValidationData {
    persisted:PersistedValidationData,
    transient:TransientValidationData
}
export interface   PersistedValidationData {

    parent_head: HeadData,
    block_number: BlockNumber,
    hrmp_mqc_heads: Vec<ITuple<[u32, Hash]>>,
    dmq_mqc_head: Hash,
    max_pov_size: u32,
}
export interface  TransientValidationData {

    max_code_size: u32,
    max_head_data_size: u32,
    balance: Balance,
    code_upgrade_allowed: Option<BlockNumber>,
    dmq_length: u32,
}
```

for the d.ts type files

and

```

    ValidationData:{
      persisted:PersistedValidationData,
      transient:TransientValidationData
    },
    PersistedValidationData:{

      parent_head: 'HeadData',
      block_number: 'BlockNumber',
      hrmp_mqc_heads: 'Vec<(Id, Hash)>',
      dmq_mqc_head: 'Hash',
      max_pov_size: 'u32',
    },
    TransientValidationData:{

      max_code_size: 'u32',
      max_head_data_size: 'u32',
      balance: 'Balance',
      code_upgrade_allowed: 'Option<BlockNumber>',
      dmq_length: 'u32',
    },
```

for js

I injected them into the `/apps/node_modules/@polkadot/types/interfaces/parachains/definitions.js` and `apps/node_modules/@polkadot/types/interfaces/parachains/types.d.ts` and also tried with interfaces/runtime but neither work.

Any idea how I should inject those new types @jacogr ?

**Comment by jacogr** [comment]
Closed in #2954

**Comment by jacogr** [comment]
Imported into apps in https://github.com/polkadot-js/apps/pull/4220
